include/zephyr/bluetooth/gatt.h

```diff
@@ -1007,15 +1007,57 @@ struct bt_gatt_notify_params {
 int bt_gatt_notify_cb(struct bt_conn *conn,
 		      struct bt_gatt_notify_params *params);
 
-/** @brief Notify multiple attribute value change.
- *
- *  This function works in the same way as @ref bt_gatt_notify_cb.
- *
- *  @param conn Connection object.
- *  @param num_params Number of notification parameters.
- *  @param params Array of notification parameters.
- *
- *  @return 0 in case of success or negative value in case of error.
+/** @brief Send multiple notifications in a single PDU.
+ *
+ *  The GATT Server will send a single ATT_MULTIPLE_HANDLE_VALUE_NTF PDU
+ *  containing all the notifications passed to this API.
+ *
+ *  All `params` must have the same `func` and `user_data` (due to
+ *  implementation limitation). But `func(user_data)` will be invoked for each
+ *  parameter.
+ *
+ *  As this API may block to wait for Bluetooth Host resources, it is not
+ *  recommended to call it from a cooperative thread or a bluetooth callback.
+ *
+ *  The peer's GATT Client must write to this device's Client Supported Features
+ *  attribute and set the bit for Multiple Handle Value Notifications before
+ *  this API can be used.
+ *
+ *  Only use this API to force the use of the ATT_MULTIPLE_HANDLE_VALUE_NTF PDU.
+ *  For standard applications, `bt_gatt_notify_cb` is preferred, as it will use
+ *  this PDU if supported and automatically fallback to ATT_HANDLE_VALUE_NTF
+ *  when not supported by the peer.
+ *
+ *  This API has an additional limitation: it only accepts valid attribute
+ *  references and not UUIDs like `bt_gatt_notify` and `bt_gatt_notify_cb`.
+ *
+ *  @param conn
+ *    Target client.
+ *    Notifying all connected clients by passing `NULL` is not yet supported,
+ *    please use `bt_gatt_notify` instead.
+ *  @param num_params
+ *    Element count of `params` array. Has to be greater than 1.
+ *  @param params
+ *    Array of notification parameters. It is okay to free this after calling
+ *    this function.
+ *
+ *  @retval 0
+ *    Success. The PDU is queued for sending.
+ *  @retval -EINVAL
+ *    - Only one parameter was passed. This API expects 2 or more.
+ *    - Not all `func` were equal or not all `user_data` were equal.
+ *    - One of the characteristics is not notifiable.
+ *    - An UUID was passed in one of the parameters.
+ *  @retval -ERANGE
+ *    - The notifications cannot all fit in a single ATT_MULTIPLE_HANDLE_VALUE_NTF.
+ *    - They exceed the MTU of all open ATT bearers.
+ *  @retval -EPERM
+ *    The connection has a lower security level than required by one of the
+ *    attributes.
+ *  @retval -ENOENT
+ *    One of the attribute handles is invalid.
```

Comment:
I think you're right. we were on the fence and had selected this because UUID lookup was still a thing. But now that I removed it, it looks like einval might be more appropriate.